switchdev_utils/config-service/switchdev_network_config

```diff
@@ -0,0 +1,548 @@
+    except Exception as e:
+        logger.error("Failed to restart openvswitch\n%s" % str(e))
+        raise e
+
+
+def write_ifcfg_files(ifcfg_interfaces_map):
+    logger.info("Writing ifcfg files for all net devices")
+    for interface, ifcfg_data in ifcfg_interfaces_map.items():
+        write_config(ifcfg_config_path(interface), ifcfg_data)
+
+
+def if_down_interface(device):
+    logger.info("Running /sbin/ifdown %s" % device)
+    try:
+        run_command('/sbin/ifdown', device)
+    except Exception as e:
+        logger.error("Failed to ifdown %s" % (device, str(e)))
+        raise e
+
+
+def if_up_interface(device):
+    logger.info("Running /sbin/ifup %s" % device)
+    try:
+        run_command('/sbin/ifup', device)
+    except Exception as e:
+        logger.error("Failed to ifup  %s" % (device, str(e)))
+        raise e
+
+
+def bring_interfaces_up(net_interfaces_list):
+    logger.info("Bringing interfaces down")
+    for interface in net_interfaces_list:
+        if_down_interface(interface)
+    logger.info("Bringing interfaces up")
+    for interface in net_interfaces_list:
+        if_up_interface(interface)
+
+
+def validate_config(sriov_pfs, bonds, bridges):
+    sriov_pfs_list = []
+    bonds_list = []
+    for sriov_pf in sriov_pfs:
+        if sriov_pf.get("name"):
+            sriov_pfs_list.append(sriov_pf.get("name"))
+        else:
+            err = "sriov pf name must be provided"
+            logger.error(err)
+            raise ValueError(err)
+        try:
+            if int(sriov_pf.get("numOfVfs")):
+                continue
+        except ValueError as value_err:
+            logger.error("numOfVfs for %s must be number" %
+                         sriov_pf.get("name"))
+            raise value_err
+        except TypeError as type_err:
+            logger.error("numOfVfs for %s must be provided" %
+                         sriov_pf.get("name"))
+    for bond in bonds:
+        if bond.get("name"):
+            bonds_list.append(bond.get("name"))
+        else:
+            err = "bond name must be provided"
+            logger.error(err)
+            raise ValueError(err)
+        if not bond.get("bondingOptions"):
+            err = "bondingOptions must be provided for bond %s" %\
+                  bond.get("name")
+            logger.error(err)
+            raise ValueError(err)
+        if (bond.get("slaves") and
+                (set(bond.get("slaves")).issubset(set(sriov_pfs_list)))):
+            continue
+        else:
+            err = "slaves of bond %s must be list of sriov_pfs" % \
+                  bond.get("name")
+            logger.error(err)
+            raise ValueError(err)
+    for bridge in bridges:
+        if not bridge.get("name"):
+            err = "bridge name must be provided"
+            logger.error(err)
+            raise ValueError(err)
+        if (bridge.get("ports") and
+                (set(bridge.get("ports")).issubset(set(sriov_pfs_list +
```

Comment:
technically if a PF is a member of a bond, it must not be a member of a bridge right ?